work/worker.go

```diff
@@ -765,6 +768,13 @@ CommitTransactionLoop:
 		if bundleIdx := builder.FindBundleIdx(bundles, txOrGen); bundleIdx != -1 {
 			targetBundle = bundles[bundleIdx]
 			numShift = len(targetBundle.BundleTxs)
+			// Skip this bundle if target is required but either:
+			// 1. The previous transaction hash doesn't match the target hash, or
+			// 2. The previous transaction failed (receipt status not successful)
+			if env.shouldDiscardBundle(targetBundle) {
+				builder.PopTxs(&incorporatedTxs, numShift, &bundles, env.signer)
```

Comment:
Could this easily break other bundles? 
e.g) 
auctionTargetTx, [auctionTx(from nodekey)], [GA.Lend1(from nodekey), GA.Approve1, GA.Swap1], [GA.Lend2(from nodekey), GA.Approve2, GA.Swap2]...
if auctionTargetTx fails, all GA.Lend will be deleted and all GA bundles will also be removed.

However, there is a cost to failing TargetTx, so this may be negligible.

Reply:
I missed this condition in `PopTxs`. The above cannot happen. I'm sorry.
https://github.com/kaiachain/kaia/pull/466/files#diff-c35437f0e3b2d97d43da03e286ede68ff96ae680ddeb7e2f1890d9de9bc7cae2R219